scripts/test-remote-dev-codex-runtime.py

```
@@ -272,6 +281,171 @@ def test_successful_publication_retains_previous_generation(self):
+                    self.assertEqual(staging.parent, fixed)
+                    self.assertFalse(staging.is_relative_to(caller_tmp))
+                    self.assertEqual(staging.stat().st_mode & 0o777, 0o711)
+                self.assertFalse(created.exists())
+                self.assertEqual(list(fixed.iterdir()), [])
+
+    def test_update_staging_cleans_after_error_and_signal(self):
+        with tempfile.TemporaryDirectory() as text:
+            fixed = self.staging_root(Path(text))
+            with mock.patch.object(self.m, "STAGING_ROOT", fixed):
+                with self.assertRaisesRegex(self.m.ManagerError, "synthetic failure"):
+                    with self.m.update_staging() as staging:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)